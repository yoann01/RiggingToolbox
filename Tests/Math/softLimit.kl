
require RiggingToolbox;

operator entry(){

    Scalar maxVal = 10.0;
    Vec2 maxValSoftening(3.0, 3.0);

    for(Integer i=0; i<20; i++){
        Scalar inval = Scalar(i);
        Scalar outval = softLimit(inval, maxVal, maxValSoftening);
        report("inval:" + inval + " outval:" + outval);
    }
}

