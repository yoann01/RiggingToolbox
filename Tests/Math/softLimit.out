inval:+0.0 outval:+0.0
inval:+1.0 outval:+1.0
inval:+2.0 outval:+2.0
inval:+3.0 outval:+3.0
inval:+4.0 outval:+4.0
inval:+5.0 outval:+5.0
inval:+6.0 outval:+6.0
inval:+7.0 outval:+7.0
inval:+8.0 outval:+7.5
inval:+9.0 outval:+8.0
inval:+10.0 outval:+8.5
inval:+11.0 outval:+9.0
inval:+12.0 outval:+9.5
inval:+13.0 outval:+10.0
inval:+14.0 outval:+10.0
inval:+15.0 outval:+10.0
inval:+16.0 outval:+10.0
inval:+17.0 outval:+10.0
inval:+18.0 outval:+10.0
inval:+19.0 outval:+10.0
